### ISSUE TITLE
Serialize subclass as baseclass

### ISSUE THREAD

**Issue by wojtpl2** [wojtpl2/ExtendedXmlSerializer#176]
Serialize subclass as baseclass doesn't work.

**Comment by wojtpl2** [comment]
I've created Issue176.cs.

**Comment by Mike-EEE** [comment]
OK cool... I'll check this out now. :)

**Comment by Mike-EEE** [comment]
Hm... it says I closed it?  Please verify on your end and close it if it works for you. 👍

**Comment by wojtpl2** [comment]
Is still does not work.

**Comment by wojtpl2** [comment]
In this case ShouldBeEquivalentTo does not work at all. They have the same problem. I'm going to create issue in their repo.

**Comment by Mike-EEE** [comment]
Ah ok... I'll check it out.  Please note that I upgraded to latest FluentAssertions in v3 branch.  There are breaking changes around `ShouldBeEquivalentTo`.

**Comment by wojtpl2** [comment]
I've created issues in FluentAssertions repo:
https://github.com/fluentassertions/fluentassertions/issues/807

In our case, we should always use `RespectingRuntimeTypes` options.:
```C#
actual.Should().BeEquivalentTo(expected, o=>o.RespectingRuntimeTypes()); //
```

**Comment by Mike-EEE** [comment]
AH... yes, that is a goooooood point.  I will try to get to this today.  If not, tomorrow. 😄

**Comment by Mike-EEE** [comment]
WOW... what a rough adventure.  Fortunately in v3 I have already laid down the groundwork to entirely re-write this entire structure and it's sooooooo much better.  In any case I am making due with what we have here in v2 and I think it's all working as expected now.  Tests are passing, etc.  But please of course feel free to add more tests if it's not working as expected with another scenario. 👍

**Comment by Mike-EEE** [comment]
Hmmm looks like we have a build error @wojtpl2:

```
Get-Nuget : Exception calling "DownloadFile" with "2" argument(s): "The operation has timed out"
At line:163 char:31
+ ... ugetPath = (Get-Nuget -toolsDir $toolsDir -nugetDownloadUrl $nugetDow ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-Nuget], MethodInvocationException
    + FullyQualifiedErrorId : WebException,Get-Nuget
 
psbuild not found, and was not downloaded successfully. sorry.
	Check your nuget.config (default path=C:\Users\appveyor\AppData\Roaming\NuGet\NuGet.config) file to ensure that nuget.org is enabled.
	You can also try changing the versionToInstall value.
	You can file an issue at https://github.com/ligershark/psbuild/issues.
```

**Comment by Mike-EEE** [comment]
Looks like tests are passing now.  Closing this issue accordingly.  Please reopen/create a new issue if something is still not working as expected.

**Comment by wojtpl2** [comment]
This issue was released in version 2.1.3. Release notes: #162
